ingestion/functions/parsing/japan/japan.py

```diff
@@ -0,0 +1,121 @@
+import json
+import os
+import sys
+from datetime import date, datetime
+
+# Layer code, like parsing_lib, is added to the path by AWS.
+# To test locally (e.g. via pytest), we have to modify sys.path.
+# pylint: disable=import-error
+if ('lambda' not in sys.argv[0]):
+    sys.path.append(
+        os.path.join(
+            os.path.dirname(os.path.dirname(os.path.abspath(__file__))),
+            'common/python'))
+import parsing_lib
+
+
+def convert_date(raw_date):
+    """
+    Convert raw date field into a value interpretable by the dataserver.
+
+    The date is listed in dd/mm/YYYY format, but the data server API will
+    assume that ambiguous cases (e.g. "05/06/2020") are in mm/dd/YYYY format.
+    """
+    date = datetime.strptime(raw_date, "%d/%m/%Y")
+    return date.strftime("%m/%d/%YZ")
+
+
+def convert_gender(raw_gender):
+    try:
+        if raw_gender["gender"] == "M":
+            return "Male"
+    except KeyError:
+        return "Unknown"
+
+    try:
+        if raw_gender["gender"] == "F":
+            return "Female"
+    except KeyError:
+        return "Unknown"
+
+def convert_age(raw_age):
+    age_range = []
+    if raw_age["ageBracket"] != -1:
+        age_range.append(raw_age["ageBracket"]),
+        age_range.append(raw_age["ageBracket"] + 9)
+    else:
+        age_range.append(""),
+        age_range.append("")
+    return age_range
+        
+
+def convert_location(raw_entry):
+    prefecture = raw_entry["detectedPrefecture"]
+    try:
+       city = raw_entry["detectedCityTown"]
+    except KeyError:
+        city = ""
+    
+    query_terms = ("Japan",)
+    location = {"country": "Japan"}
+    if prefecture:
+        location["administrativeAreaLevel1"] = prefecture
+        query_terms = (prefecture,) + query_terms
+    if city:
+        location["administrativeAreaLevel2"] = city
```

Comment:
Really? am I reading that table wrong... To me it looks as though prefectures are level1, cities level2, and municipalities level3

Reply:
Seems to be a little nuanced. Outside of Tokyo, admin2 is comprised of "28 sub-prefectures" and "15 designated cities"; admin3 is "1700+ municipalities," where municipality ~= town or city.

Looking at the API data, I can see some entries for `detectedCityTown` that are admin3 -- like Yuasa. Alternatively, I can see some that are admin2 -- like Nagoya, which is a designated city. Within Tokyo looks a bit complicated as well.

Let me poke around and see if I can find someone to weigh in on how this data is typically encoded.

Reply:
I agree, "cityTown" is "shichou/市町" which corresponds to admin3 when outside Tokyo according to your wikipedia article.
![image](https://user-images.githubusercontent.com/1255432/89500765-88626b00-d7c2-11ea-9f4d-05311930ad99.png)

Within tokyo though we can only get up to admin 2 it seems, so perhaps check for the detected prefecture being "Tokyo" and then put the detected cititown in admin 2 instead of admin3 in that case?

Reply:
Great, thanks for your help guys.